src/Symfony/Component/Serializer/Instantiator/InstantiatorResult.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Instantiator;
+
+/**
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+class InstantiatorResult
+{
+    private $object;
+    private $data;
+    private $context;
+    private $error;
+
+    public function __construct(?object $object, array $data, array $context, string $error = null)
```

Comment:
Why not typing the last parameter as `?string $error = null` ? Honest question, I don't have an opinion about it.

Reply:
There is actually no case we set `$error` when it's null, so it's not needed there.